csharp/extractor/Semmle.Extraction.CSharp/Entities/Symbol.cs

```diff
@@ -136,16 +136,32 @@ protected void PopulateMetadataHandle(TextWriter trapFile)
                 trapFile.metadata_handle(this, Location, MetadataTokens.GetToken(handle.Value));
         }
 
+        static System.Reflection.PropertyInfo GetPropertyInfo(object o, string name) =>
```

Comment:
Should we optimize for better performance? We could cache the `PropertyInfo` by type and save that bit of the reflection. That's an easy win.

Reply:
I guess we could, but I would like to move this logic into QL, so not sure if it is worthwhile. We could perhaps try it in a follow-up PR?